src/input/index.es

```diff
@@ -3,25 +3,49 @@ import PropTypes from 'prop-types';
 import omit from 'lodash/omit';
 import classNames from 'classnames';
 
+import { getDummyName } from './utils';
 import TextInputComponent from './base';
 
 
 class Input extends TextInputComponent {
+  componentDidMount() {
+    super.componentDidMount();
+    this.generateDummyName();
+  }
+
+  componentDidUpdate(prevProps) {
```

Comment:
remove this section, we need neither props tracking nor state. way too complicated